lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -9131,7 +9131,8 @@ namespace Js
                 }
                 else
                 {
-                    JS_REENTRANT(jsReentLock, obj->DeleteItem(toIndex, PropertyOperation_ThrowOnDeleteIfNotConfig));
+                    JavascriptOperators::OP_DeleteElementI(obj, JavascriptNumber::ToVar(toIndex, scriptContext), scriptContext, PropertyOperation_ThrowOnDeleteIfNotConfig);
```

Comment:
Don't we still need to wrap this in JS_REENTRANT?

Reply:
For more context, we explicitly mark any lines in here that are allowed to call back into script (it would be a bug if any line without such a marking could call script). This pattern is useful because we can assume that certain things don't change as long as we haven't called into script (for instance, an instance of JavascriptArray hasn't magically transformed itself into an instance of ES5Array). The object here could be a Proxy with a deleteProperty handler, in which case this deletion operation would call script.

---
In reply to: [173333445](https://github.com/Microsoft/ChakraCore/pull/4790#discussion_r173333445) [](ancestors = 173333445)

Reply:
Thank you for the explanations! I'm reverting this file but I might need to follow up offline to understand better about the Proxies and reentrancy :)

---
In reply to: [173337067](https://github.com/Microsoft/ChakraCore/pull/4790#discussion_r173337067) [](ancestors = 173337067,173333445)